lib/filters/global/parsoid.js

```diff
@@ -100,6 +100,61 @@ function transform(parsoidHost, from, to) {
     };
 }
 
+function transformRevision(parsoidHost, from, to) {
+    return function (restbase, req) {
+
+        var domain = req.params.domain;
+        var key    = req.params.key;
+        var rev    = req.params.revision;
+
+        var original = {
+            revid: rev
+        };
+
+        function get(format) {
+            return restbase.get({ uri: '/v1/' + domain + '/pages/' + key + '/' + format + '/' + rev })
+            .then(function (res) {
+                if (res.body &&
+                    res.body.headers && res.body.headers['content-type'] &&
+                    res.body.body) {
+                    return {
+                        status: 200,
+                        headers: {
+                            'content-type': res.body.headers['content-type']
+                        },
+                        body: res.body.body
+                    };
+                } else {
+                    return {
+                        status: 404
+                    };
+                }
+            })
+            .then(function (res) {
+                if (res.status === 200) {
+                    original[format] = {
```

Comment:
It might be slightly nicer / easier to follow to return the result rather than using a side-effect to fill out original. The assembly would then be based on the array returned from Promise.all() below.